pulpcore/pulpcore/app/viewsets/base.py

```diff
@@ -158,3 +161,58 @@ class CreateDestroyReadNamedModelViewSet(mixins.CreateModelMixin,
 
     """
     pass
+
+
+class AsyncUpdateMixin(object):
+    """
+    Provides an update method that dispatches a task with reservation for a repository
+    """
+    def update(self, request, pk, **kwargs):
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        app_label = instance._meta.app_label
+        async_result = tasks.base.general_update.apply_async_with_reservation(
+            instance._meta.db_table, pk,
```

Comment:
Ah does this mean that importers will reserve the importer table, and publisher will reserve the publisher table? Is this the correct behavior (instead of both reserving the repository table)?

Reply:
Yes, that is correct. With introduction of versioned repositories, it does not make sense to lock on a particular repository. Syncs will lock on an importer also. Publishes will lock on publisher, and repository version.